### PR TITLE
Template for RunOfOnes challenge

### LINKED ISSUE #91: [Challenge] Runs of Ones
# Runs of Ones

Given a positive integer `n`, determine the length of the longest run of `1`'s in the binary representation of that integer.

## Business Rules/Errata

- A run of `1`'s is any series of one or more `1`'s in a row, with no `0`'s in between. The first four characters in `11110101` are a run of `1`'s.
- There's no requirement related to what data type(s) should be used to store the binary representation of `n`.
- Target time complexity: O(n) (linear)
- Target space complexity: O(1) (constant)

## Examples

```
longest_binary_run(156);   // 3
longest_binary_run(1979);  // 4
longest_binary_run(2731);  // 2
longest_binary_run(2185);  // 1
```

### PR DESCRIPTION
**Pull request by parhaml**
closes #91 

PLEASE please check my work.

I copied everything from a previous `java` challenge and tried to frankenstien it together. Thanks!

### CODE REVIEW

**Comment by ericwburden** [comment]
It looks like you copied from the Intervening Verbiage setup (which is an older Java setup). I'd recommend looking at the Find Three Largest Numbers (06/01) challenge, which simplifies the setup and uses symlinks for the test script. My other recommendation would be to include a test for 0 as an edge case.

File: challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    /**
+     * Second test case
+     */
+    @Test
+    @DisplayName("For the number: 1979")
+    public void testCaseTwo() {
+        int result = testClass.longest_binary_run(1979);
+        assertEquals(3, result);
```
Comment (ericwburden) [inline challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java:41]: should be `assertEquals(4, result);`

File: challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    /**
+     * Third test case
+     */
+    @Test
+    @DisplayName("For the number: 2731")
+    public void testCaseThree() {
+        int result = testClass.longest_binary_run(2731);
+        assertEquals(3, result);
```
Comment (ericwburden) [inline challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java:50]: should be `assertEquals(2, result);`

File: challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    /**
+     * Fourth test case
+     */
+    @Test
+    @DisplayName("For the number: 2185")
+    public void testCaseFour() {
+        int result = testClass.longest_binary_run(2185);
+        assertEquals(3, result);
```
Comment (ericwburden) [inline challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java:60]: should be `assertEquals(1, result);`

File: challenges/2021-06-29-run-of-ones/test-it.sh [hunk 4]
```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
```
Comment (ericwburden) [inline challenges/2021-06-29-run-of-ones/test-it.sh:1]: The newer version of this challenge setup uses a symlink to the testing script.

File: challenges/2021-06-29-run-of-ones/src/main/java/io/codeconnector/codedojo/RunOfOnes.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package io.codeconnector.codedojo;
```
Comment (ericwburden) [inline challenges/2021-06-29-run-of-ones/src/main/java/io/codeconnector/codedojo/RunOfOnes.java:1]: The newer version of the Java challenge setup includes the code and tests in a single file/class.

File: challenges/2021-06-29-run-of-ones/mob.java [hunk 6]
```diff
@@ -0,0 +1 @@
+src/main/java/io/codeconnector/codedojo/InterveningVerbiage.java
```
Comment (ericwburden) [inline challenges/2021-06-29-run-of-ones/mob.java:1]: This symlink is broken.

File: challenges/2021-06-29-run-of-ones/challenge.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+## Examples
+
+```
+longest_binary_run(156);   // 3
+longest_binary_run(1979);  // 4
+longest_binary_run(2731);  // 2
+longest_binary_run(2185);  // 1
+```
+
+## Tackling This Challenge
```
Comment (ericwburden) [inline challenges/2021-06-29-run-of-ones/challenge.md:21]: This section includes some references to the challenge it was copied from. The updated Java setup format includes a briefer 'Tackling This Challenge' section with more generic comments.

**Review by ericwburden — CHANGES_REQUESTED** [review]
Most of my comments are pretty much just beating a dead horse (about using a more recent Java challenge as the setup template), but do check the comments on the tests (they currently all pass with a result of `3`).

**Comment by parhaml** [comment]
> Most of my comments are pretty much just beating a dead horse (about using a more recent Java challenge as the setup template), but do check the comments on the tests (they currently all pass with a result of `3`).
Oooh. Nice catch. I did that to ensure the passing shell stuff would work and forgot to change it back.


I'll definitely look at the other set-up. Thanks!

File: challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java (hunk 1 above)
Comment (parhaml) [reply challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java:41]: updated!

File: challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java (hunk 3 above)
Comment (parhaml) [reply challenges/2021-06-29-run-of-ones/src/test/java/io/codeconnector/codedojo/RunOfOnesTest.java:60]: updated

**Comment by parhaml** [comment]
@ericwburden this has been updated

**Review by ericwburden — APPROVED** [review]
Looks good to me. I made one change to the comments in the RunOfOnes.java file.

**Comment by parhaml** [comment]
> Looks good to me. I made one change to the comments in the RunOfOnes.java file.

Doh. Nice
